tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_gemv_kk.hpp

```diff
@@ -0,0 +1,228 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+
+namespace KokkosKernelsSTD {
+
+namespace gemvimpl{
+template<class Accessor>
+double get_scaling_factor(Accessor) { return 1.0; }
+
+template<class Accessor, class S>
+auto get_scaling_factor(std::experimental::linalg::accessor_scaled<Accessor,S> a) {
+  return a.scale_factor();
+}
+} //end anon namespace
+
+//
+// overwriting gemv: y = Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+         class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+void matrix_vector_product(kokkos_exec<ExeSpace> /*kexe*/,
+			   std::experimental::mdspan<
+			     ElementType_A,
+			     std::experimental::extents<numRows_A, numCols_A>,
+			     Layout_A,
+			     std::experimental::default_accessor<ElementType_A>
+			   > A,
+			   std::experimental::mdspan<
+			     ElementType_x,
+			     std::experimental::extents<ext_x>,
+			     Layout_x,
+			     std::experimental::default_accessor<ElementType_x>
+			   > x,
+			   std::experimental::mdspan<
+			     ElementType_y,
+			     std::experimental::extents<ext_y>,
+			     Layout_y,
+			     std::experimental::default_accessor<ElementType_y>
+			   > y)
+{
+
+  // constraints
+  static_assert(A.rank() == 2);
+  static_assert(x.rank() == 1);
+  static_assert(y.rank() == 1);
+
+  // preconditions
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+
+  // this print is detected in the tests
+#if defined KOKKOS_STDBLAS_ENABLE_TESTS
+  std::cout << "overwriting_matrix_vector_product: kokkos impl\n";
```

Comment:
I've made this comment on other PRs related to this one: Would you consider encapsulating logging, so that mdspan files aren't exposed to `#include <iostream>`?
```c++
extern void logKokkos(const std::string_view functionName);
```
with `logKokkos` defined in a .cpp file somewhere, would make it possible to log in different ways, without needing to change existing code.